app/views/layouts/application.html.erb

```diff
@@ -9,6 +9,13 @@
   </head>
 
   <body>
+    <% if notice %>
```

Comment:
They can both happen at the same time, so I think this format works - you would want them both to appear if both are valid (there are 4 possible outcomes: notice && alert, notice && !alert, !notice && alert, !notice && !alert)